### PR TITLE
Create a vpc for the ec2 instance

### PR DESCRIPTION
**Pull request by dan-petty**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,5 +1,6 @@
 provider "aws" {
   version = "~> 2"
+  region = var.region
```
Comment (AaronNBrock) [inline main.tf:3]: I usually prefer leaving this unspecified and using the environment variable `AWS_DEFAULT_REGION`, it allows more options of authentication: https://www.terraform.io/docs/providers/aws/index.html

File: main.tf [hunk 2]
```diff
@@ -61,3 +71,141 @@ resource "aws_security_group" "ssh" {
     Name = "${var.security_group_name}${var.suffix}"
   }
 }
+
+resource "aws_security_group" "vpc_endpoints" {
+  name = "${var.app_name}-${var.env_name}${var.suffix}-vpc-endpoints-sg"
```
Comment (AaronNBrock) [inline main.tf:76]: The suffix should be at the very end of the resource, I follow this pattern so iam resource selectors can be done via `*-my-suffix`

Comment (AaronNBrock) [inline main.tf:76]: Also, if we want to add a `var.app_name`/`env_name` into names it'd be good to use it consistently across all the names.

Also, maybe these should be added as tags?

File: main.tf (hunk 1 above)
Comment (dan-petty) [reply main.tf:3]: No worries, was just working off terraform validate output. I'm sure it is fine if AWS_DEFAULT_REGION is set.
```
terraform validate

Error: Missing required argument

  on main.tf line 1, in provider "aws":
   1: provider "aws" {

The argument "region" is required, but no definition was found.
```

File: main.tf (hunk 2 above)
Comment (dan-petty) [reply main.tf:76]: I generally use resource type as the name suffix, as in this naming scheme:
https://confluence.huit.harvard.edu/display/CLA/Cloud+Resource+Naming+Conventions

That being said I'm not particularly attached to any one way of doing this.

Comment (dan-petty) [reply main.tf:76]: That's what I did for the resources I added, but I didn't want to clobber your variables ticket work so left other resources unchanged.

File: main.tf (hunk 1 above)
Comment (AaronNBrock) [reply main.tf:3]: Fair enough, actually defining it here is probably a more declarative way to handle it anyhow.

File: main.tf (hunk 2 above)
Comment (AaronNBrock) [reply main.tf:76]: Neat, I have no strong feelings on this either so I say we stick to that doc.  This does mean we need to update the already existing names, but maybe lets do that in a new pull request.

**Comment by AaronNBrock** [comment]
Fixed with #2

**Pull request closed without merging**